driver.py

```diff
@@ -0,0 +1,79 @@
+import unzip
+import extract
+import poster
+import argparse
+import nxttxt
+from pathlib import Path
+
+
+def unzip_drive(unzip_input: str, cache: str, output):
+    cache_path = Path(cache)
+    if not cache_path.exists():
+        cache_path.mkdir()
+    else:
+        nxttxt.clear_directory(cache_path, False)
+
+    auth = unzip.authenticate_instructional_validity(unzip_input, output)
+    if auth[0]:
+        unzip.unzip_file(unzip_input, cache_path, True)
+    else:
+        nxttxt.clear_directory(cache_path, True)
+        print(auth[1])
+        raise auth[2]
+
```

Comment:
Is auth always going to be a list of 3? why print 1 and raise 2?

Reply:
Sometimes! In every auth array, the first element is whether user inputs are valid and can be used by the script, the second is the error message, and the third is the corresponding more general exception to used if this is being run inside a larger application. The second and third elements only exist if the first one is equal to false and likewise that is the only case in which they will be read. Do you think I should change this to be a dict? Do you have another structure you would recommend?